ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/BlockPropagationManager.java

```diff
@@ -77,12 +78,14 @@
       final EthContext ethContext,
       final SyncState syncState,
       final PendingBlocks pendingBlocks,
-      final LabelledMetric<OperationTimer> ethTasksTimer) {
+      final LabelledMetric<OperationTimer> ethTasksTimer,
+      final BlockBroadcaster<C> blockBroadcaster) {
```

Comment:
This field is always overwritten, no need for it in the constructor.

Reply:
as I wrote in the description, it's easier to test if it's in the constructor. do you suggest making an explicit getter and setter?

Reply:
The fact it is overwritten is the real concern and never used.  If it's fixed later then it can stay.

Reply:
yeah- good point about it getting overwritten- mock passed to constructor never getting used 😬- so then we should have a getter and setter then?

Reply:
No, passing it into the constructor is good, but `BlockBroadcaster` shouldn't take `Bock` into its constructor, only pass in the block to broadcast to the method that actual broadcasts it.  That way you can reuse the same `BlockBroadcaster` instance to broadcast multiple blocks and never need to create a new one in `BlockPropagationManager`.